pkg/executor/infoschema_reader.go

```diff
@@ -3802,8 +3746,64 @@ func (e *memtableRetriever) setDataFromIndexUsage(ctx context.Context, sctx sess
 	return nil
 }
 
-func (e *memtableRetriever) setDataForClusterIndexUsage(ctx context.Context, sctx sessionctx.Context, schemas []model.CIStr) error {
-	err := e.setDataFromIndexUsage(ctx, sctx, schemas)
+func (e *memtableRetriever) setDataFromIndexUsage(ctx context.Context, sctx sessionctx.Context) error {
+	dom := domain.GetDomain(sctx)
+	rows := make([][]types.Datum, 0, 100)
+	checker := privilege.GetPrivilegeManager(sctx)
+	extractor, ok := e.extractor.(*plannercore.InfoSchemaIndexUsageExtractor)
+	if !ok {
+		return errors.Errorf("wrong extractor type: %T, expected InfoSchemaIndexUsageExtractor", e.extractor)
+	}
+	if extractor.SkipRequest {
+		return nil
+	}
+
+	schemas := extractor.ListSchemas(e.is)
+	for _, schema := range schemas {
+		tbls, err := extractor.ListTables(ctx, schema, e.is)
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		for _, tbl := range tbls {
+			if checker != nil && !checker.RequestVerification(
+				sctx.GetSessionVars().ActiveRoles,
+				schema.L, tbl.Name.L, "", mysql.AllPrivMask) {
+				continue
+			}
+
+			idxs := extractor.ListIndexes(tbl)
+			for _, idx := range idxs {
+				row := make([]types.Datum, 0, 14)
```

Comment:
Does it support the index_name filter?

Reply:
select ... from index_usage where table_schema = 'xxx' and index_name = 'yyy'

Reply:
Yes, it does.

Reply:
I see, it's inside the ListIndexes() function.